spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarAdministration.java

```diff
@@ -161,39 +164,52 @@ private void createOrModifyTopicsIfNeeded(Collection<PulsarTopic> topics) {
-		Map<String, List<PulsarTopic>> topicsPerNamespace = getTopicsPerNamespace(topics);
+		var topicsPerNamespace = getTopicsPerNamespace(topics);
 
 		topicsPerNamespace.forEach((namespace, requestedTopics) -> {
-			Set<PulsarTopic> topicsToCreate = new HashSet<>();
-			Set<PulsarTopic> topicsToModify = new HashSet<>();
+			var topicsToCreate = new HashSet<PulsarTopic>();
+			var topicsToModify = new HashSet<PulsarTopic>();
 
 			try {
-				List<String> existingTopicsInNamespace = admin.topics().getList(namespace);
+				var existingTopicsInNamespace = admin.topics().getList(namespace);
 
-				for (PulsarTopic topic : requestedTopics) {
+				for (var topic : requestedTopics) {
+					var topicName = topic.getFullyQualifiedTopicName();
 					if (topic.isPartitioned()) {
-						List<String> matchingPartitions = getMatchingTopicPartitions(topic, existingTopicsInNamespace);
+						if (existingTopicsInNamespace.contains(topicName)) {
+							throw new IllegalStateException(
+									"Topic '%s' already exists un-partitioned. Needs to be deleted first."
+											.formatted(topicName));
+						}
+						var matchingPartitions = getMatchingTopicPartitions(topic, existingTopicsInNamespace);
 						if (matchingPartitions.isEmpty()) {
-							this.logger.debug(() -> "Topic " + topic.getFullyQualifiedTopicName() + " does not exist.");
+							this.logger.debug(() -> "Topic '%s' does not yet exist - will add".formatted(topicName));
 							topicsToCreate.add(topic);
 						}
 						else {
-							int numberOfExistingPartitions = matchingPartitions.size();
+							var numberOfExistingPartitions = matchingPartitions.size();
 							if (numberOfExistingPartitions < topic.numberOfPartitions()) {
-								this.logger.debug(() -> "Topic " + topic.getFullyQualifiedTopicName() + " found with "
-										+ numberOfExistingPartitions + " partitions.");
+								this.logger.debug(() -> "Topic '%s' found with %d partitions - will update to %d"
+										.formatted(topicName, numberOfExistingPartitions, topic.numberOfPartitions()));
 								topicsToModify.add(topic);
 							}
 							else if (numberOfExistingPartitions > topic.numberOfPartitions()) {
-								throw new IllegalStateException("Topic " + topic.getFullyQualifiedTopicName()
-										+ " found with " + numberOfExistingPartitions
-										+ " partitions. Needs to be deleted first.");
+								throw new IllegalStateException(
+										"Topic '%s' found w/ %d partitions - can't shrink to %d. Needs to be deleted first"
+												.formatted(topicName, numberOfExistingPartitions,
+														topic.numberOfPartitions()));
 							}
 						}
 					}
 					else {
-						if (!existingTopicsInNamespace.contains(topic.getFullyQualifiedTopicName())) {
-							this.logger.debug(() -> "Topic " + topic.getFullyQualifiedTopicName() + " does not exist.");
+						var matchingPartitions = getMatchingTopicPartitions(topic, existingTopicsInNamespace);
```

Comment:
This is part two of the change: "If there is already a partitioned topic out there - barf"